Remove EventSourceAttribute usages when EventSource feature is disabled

marek-safar opened the pull request.

marek-safar committed:
Remove EventSourceAttribute usages when EventSource feature is disabled

eerhardt commented on src/libraries/System.Private.CoreLib/src/ILLink/ILLink.LinkAttributes.Shared.xml:
```diff
@@ -30,4 +30,10 @@
   </assembly>
+
+  <assembly fullname="System.Private.CoreLib" feature="System.Diagnostics.Tracing.EventSource.IsSupported" featurevalue="false">
+    <type fullname="System.Diagnostics.Tracing.EventSourceAttribute">
```
What about other attributes:

* EventAttribute
* EventDataAttribute
* EventFieldAttribute
* EventIgnoreAttribute
* NonEventAttribute

?

marek-safar committed:
Add also event information attributes

They should not be present on the ETW types but it could be used in
code which is not guarded with System.Diagnostics.Tracing.EventSource.IsSupported check

eerhardt reviewed (approved):
LGTM
